zap_text_core.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import "go.uber.org/zap/zapcore"
+
+// NewTextCore creates a Core that writes logs to a WriteSyncer.
+func NewTextCore(enc *textEncoder, ws zapcore.WriteSyncer, enab zapcore.LevelEnabler) zapcore.Core {
+	return &textIOCore{
+		LevelEnabler: enab,
+		enc:          enc,
+		out:          ws,
+	}
+}
+
+// textIOCore is a copy of zapcore.ioCore that only accept *textEncoder
+// it can be removed after https://github.com/uber-go/zap/pull/685 be merged
+type textIOCore struct {
+	zapcore.LevelEnabler
+	enc *textEncoder
+	out zapcore.WriteSyncer
+}
+
+func (c *textIOCore) With(fields []zapcore.Field) zapcore.Core {
+	clone := c.clone()
+	// it's different to ioCore, here call textEncoder#addFields to fix https://github.com/pingcap/log/pull/4
```

Comment:
Incorrect comment? This PR is https://github.com/pingcap/log/pull/4, and this comment claims that a function introduced in this PR is aimed to fix https://github.com/pingcap/log/pull/4 😂

Reply:
:rofl: